internal/controller/snapshot/snapshot_adapter_test.go

```diff
@@ -1575,6 +1575,119 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 		})
 	})
 
+	When("snapshot is older than last build", func() {
+		It("ensures that snapshot as is marked with correct auto-release message", func() {
+			// CREATE COMPONENT WITH NEWER BUILD TIME
+			componentWithNewerBuild := &applicationapiv1alpha1.Component{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-component-older",
+					Namespace: "default",
+					Annotations: map[string]string{
+						gitops.BuildPipelineLastBuiltTime: "1703123456", // Newer timestamp
+					},
+				},
+				Spec: applicationapiv1alpha1.ComponentSpec{
+					ComponentName:  "test-component-older",
+					Application:    hasApp.Name,
+					ContainerImage: sample_image + "@" + sampleDigest,
+					Source: applicationapiv1alpha1.ComponentSource{
+						ComponentSourceUnion: applicationapiv1alpha1.ComponentSourceUnion{
+							GitSource: &applicationapiv1alpha1.GitSource{
+								URL:      SampleRepoLink,
+								Revision: sourceRepoRef,
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, componentWithNewerBuild)).Should(Succeed())
```

Comment:
This resource shouldn't need to be created on the control plane since it will be [mocked in the adapter context](https://github.com/konflux-ci/integration-service/pull/1289/files#diff-1693218babdbb876adbb35bdb174a45d78351087e5434f1e4966cdae2df32a27R1654-R1671).